static/src/js/hooks/useRevisionsQuery.js

```diff
@@ -0,0 +1,133 @@
+import { useEffect, useState } from 'react'
+import { isEmpty } from 'lodash-es'
+import { useLazyQuery } from '@apollo/client'
+
+import conceptTypeQueries from '../constants/conceptTypeQueries'
+import { GET_COLLECTION_REVISIONS } from '../operations/queries/getCollectionRevisions'
+import conceptTypes from '../constants/conceptTypes'
+
+const useRevisionsQuery = ({
+  type,
+  conceptId: revisionConceptId,
+  limit,
+  offset,
+  sortKey
+}) => {
+  const [revisions, setRevisions] = useState({})
+  const [error, setError] = useState()
+  const [loading, setLoading] = useState(false)
+
+  // Sets a 'published' flag for records that are published
+  const formatRevisionResults = (results) => {
+    const { items, count, __typename } = results
+
+    const flaggedItems = items.map((item) => {
```

Comment:
I think this might break if the sortKey was ever changed. I don't think we can guarantee that the last item in the list is the most recent revision in that case. It might need to be derived by looking at the published date.

One way to do this would be to first run a `reduce` on the items where you can compare the dates and find the most recent one, like in this [example](https://stackoverflow.com/a/57731226). Then use that value to designate that item in the list as published.